pilot/pkg/model/push_context.go

```diff
@@ -152,6 +152,12 @@ type PushRequest struct {
 	// If this is present, then only proxies that import this namespace will get an update
 	TargetNamespaces map[string]struct{}
 
+	// TargetProxies contains a list of proxies that should be pushed to.
+	// The key is clusterID + proxy ip address.
+	// If this is empty, then all proxies will get an update.
+	// If this is present, then only proxies in the list will get an update.
+	TargetProxies map[string]struct{}
+
```

Comment:
do we need target namespaces and target proxies? cant we just have target proxies alone such that
 we have a map like this:

targetProxies:
` clusterID -> [namespaces -> [proxies] ]`
if map is empty, we push to all
if proxies is empty, we push to all proxies in that namespace in that cluster [if clusterId is *, then we push to all proxies in that namespace across all clusters].
if cluster Id is present, then we push only to proxies with that clusterID
and so on.

Reply:
The `TargetNamespaces` is to specify the updated services.  However, the `TargetProxies` specifies the proxies that need a push.

Reply:
ugh.. then can we rename the TargetNamespaces to UpdatedNamespaces ?